core/src/test/scala/weaponregex/parser/ParserTest.scala

```diff
@@ -27,14 +26,14 @@ trait ParserTest {
     val parsedTree = Parser(pattern, flags, parserFlavor)
 
     assert(clue(parsedTree) match {
-      case Failure(exception: RuntimeException) => exception.getMessage.startsWith(ErrorMessage.parserErrorHeader)
-      case _                                    => false
+      case Left(msg) => msg.startsWith(ErrorMessage.parserErrorHeader)
+      case _         => false
     })
   }
 
   test("Parse concat of characters") {
     val pattern = "hello"
-    val parsedTree = Parser(pattern, parserFlavor).get.to[Concat]
+    val parsedTree = Parser(pattern, parserFlavor).getOrElse(fail("Failed to parse")).to[Concat]
```

Comment:
Should this (and the others) relay the error message from the parser instead of a new one?

Reply:
Good point! I've updated it to use the error message

Reply:
Really nice trick with `either.fold(fail(_), identity)` 🧐